### DISCUSSION TITLE
Bootloader option to load from SD?

### DISCUSSION THREAD

**Discussion by u77345** [rsta2/circle#487] [Development]
I'd like to add an option to the example 38 bootloader, that boots an image from the microsd instead of attempting the network loading, if a GPIO is pulled low. Otherwise, it loads from the network. Is there any concern doing so, or perhaps an example loading the kernel from the filesystem?

**Comment by rsta2** [comment]
I think this should work and can be implemented as an additional task in *sample/38-bootloader*. I do not have an example by hand, but would prepare one, if you need help. I would need to know, on which RPi model it will run, so that I can test it on the right model.

**Reply by u77345** [reply]
The target board is "Raspberry Pi Model B R2". Thanks for offering support!

**Comment by rsta2** [comment]
You find the source code of the extended bootloader appended. Configuration is in *kernel.h* and *kernel.cpp*. By default the button must be connected from GPIO18 (SoC number) to GND. The kernel image to be loaded is *mykernel.img* in the root directory of the SD card. You can define `NO_NET_BOOT` to disable the booting functions from network.

[bootloader-extended.zip](https://github.com/user-attachments/files/17350792/bootloader-extended.zip)

**Comment by u77345** [comment]
Thanks Rene, all 3 boot modes worked beautifully.

**Reply by rsta2** [reply]
I'm glad, that it helped.
